### PR TITLE
net-misc/tigervnc: version bump 1.7.0 #593464

### PR DESCRIPTION
**Pull request by khumarahn**

### CODE REVIEW

**Comment by Amynka** [comment]
@gentoo/proxy-maint

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		else
+			rm "${D}"/usr/$(get_libdir)/xorg/modules/extensions/libvnc.la || die
+		fi
+
+		newconfd "${FILESDIR}"/${PN}.confd ${PN}
+		newinitd "${FILESDIR}"/${PN}.initd ${PN}
+		systemd_douserunit contrib/systemd/user/vncserver@.service
+	else
+		cd "${D}" || die
+		for f in vncserver vncpasswd x0vncserver vncconfig; do
```
Comment (monsieurp) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: This is a nitpick but could make `f` local? Thanks.

Comment (khumarahn) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: this makes sense. changed.

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sh ~sparc ~x86"
+IUSE="+drm gnutls java +opengl pam server +xorgmodule"
+
+RDEPEND="virtual/jpeg:0
+	sys-libs/zlib
+	>=x11-libs/libXtst-1.0.99.2
+	>=x11-libs/fltk-1.3.1
+	gnutls? ( net-libs/gnutls )
```
Comment (SoapGentoo) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: `:=`

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+		>=x11-libs/xtrans-1.3.3
+		>=x11-proto/dri2proto-2.8
+		opengl? ( >=media-libs/mesa-10.3.4-r1 )
+	)"
+
+CMAKE_IN_SOURCE_BUILD=1
+
+src_prepare() {
+	if use server ; then
+		cp -r "${WORKDIR}"/xorg-server-${XSERVER_VERSION}/. unix/xserver
```
Comment (SoapGentoo) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: missing `|| die`

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+	fi
+
+	eapply "${WORKDIR}"/patches/010_libvnc-os.patch
+	eapply "${WORKDIR}"/patches/030_manpages.patch
+	eapply "${WORKDIR}"/patches/055_xstartup.patch
```
Comment (SoapGentoo) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: put these in `PATCHES`, and let `default` call `eapply`

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+src_install() {
+	cmake-utils_src_install
+
+	newicon "${DISTDIR}"/tigervnc.png vncviewer.png
+	make_desktop_entry vncviewer vncviewer vncviewer Network
+
+	if use server ; then
+		emake -C unix/xserver/hw/vnc DESTDIR="${D}" install
+		if ! use xorgmodule; then
+			rm -r "${D}"/usr/$(get_libdir)/xorg || die
```
Comment (SoapGentoo) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: double-slashing, plus prefer ED. Make this:
`rm -r "${ED%/}"/usr/$(get_libdir)/xorg || die`

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+		else
+			rm "${D}"/usr/$(get_libdir)/xorg/modules/extensions/libvnc.la || die
```
Comment (SoapGentoo) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: same here

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+		fi
+
+		newconfd "${FILESDIR}"/${PN}.confd ${PN}
+		newinitd "${FILESDIR}"/${PN}.initd ${PN}
+		systemd_douserunit contrib/systemd/user/vncserver@.service
+	else
+		local f
+		cd "${D}" || die
```
Comment (SoapGentoo) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: `ED` rather than `D`

**Comment by khumarahn** [comment]
@SoapGentoo thank you for the feedback. I added a commit with your changes. Can you check this is what you meant?

**Comment by khumarahn** [comment]
anybody? bump?

**Comment by khumarahn** [comment]
ping

**Comment by khumarahn** [comment]
@monsieurp @SoapGentoo maybe if you review the changes you requested, this PR will go somewhere?

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+src_install() {
+	cmake-utils_src_install
+
+	newicon "${DISTDIR}"/tigervnc.png vncviewer.png
+	make_desktop_entry vncviewer vncviewer vncviewer Network
+
+	if use server ; then
+		emake -C unix/xserver/hw/vnc DESTDIR="${ED}" install
+		if ! use xorgmodule; then
+			rm -r "${ED}"usr/$(get_libdir)/xorg || die
```
Comment (SoapGentoo) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: can you make this `"${ED%/}"/` instead please? See https://devmanual.gentoo.org/ebuild-writing/variables/ "Trailing Slashes in Variables"

File: net-misc/tigervnc/tigervnc-1.7.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+		else
+			rm "${ED}"usr/$(get_libdir)/xorg/modules/extensions/libvnc.la || die
```
Comment (SoapGentoo) [inline net-misc/tigervnc/tigervnc-1.7.0.ebuild]: same here

**Comment by khumarahn** [comment]
@SoapGentoo Oh, I see now the rationale for removing the trailing slash. I overwrote the last commit.

**Comment by khumarahn** [comment]
@SoapGentoo ok, rebased :-)

**Comment by SoapGentoo** [comment]
@khumarahn thanks, I'll merge it later

**Pull request closed without merging**